sycl/doc/SYCLPluginInterface.md

```diff
@@ -0,0 +1,178 @@
+# The SYCL Runtime Plugin Interface.
+
+
+## Overview
+The SYCL Runtime Plugin Interface (PI) is the interface layer between
+device-agnostic part of the SYCL runtime and the device-specific runtime layers
+which control execution on the device. It employs the “plugin” mechanism to bind
+to the device specific runtime layers similarly to what is used by libomptarget
+or OpenCL.
+
+![PI in SYCL runtime architecture](images/SYCL_RT_arch.svg)
+
+The plugin interface and the discovery process behind it allows to dynamically
+plug in implementations based on OpenCL and “native” runtime for a particular
+device – such as OpenCL for
+FPGA devices or native runtimes for GPUs. Implementations of the PI are
+“plugins” - dynamic libraries or shared objects which expose a number of entry
+points implementing the PI interface. The SYCL runtime collects those function
+pointers into a PI interface dispatch table - one per plugin - and uses this
+table to dispatch to the device(s) covered by the corresponding plugin. Further
+sections give complete descriptions of all entry points as well as the dispatch
+table.
+
```

Comment:
This document does not five complete description if all PI interfaces for now. So I suggest just to remove this last sentence.

Reply:
ok